server/pbench/bin/pbench-index

```diff
@@ -1,247 +1,2671 @@
+                    alias_name = _aliases[fname]
+                except KeyError:
+                    # Ignore .csv files for which we don't have a handler,
+                    # after checking to see if they might have an alias
+                    # name.
+                    #print("WARNING: no .csv handler for %s (%s)\n" % (fname, p), file=sys.stdout)
+                    continue
+                else:
+                    for rec in handler['patterns']:
+                        if rec['pattern'].match(alias_name):
+                            handler_rec = rec
+                            break
+                    else:
+                        # Ignore .csv files for which we don't have a handler
+                        #print("WARNING: no .csv handler for %s (%s, %s)\n" % (alias_name, fname, p), file=sys.stdout)
+                        continue
+            assert handler_rec is not None
+            datafile = _dict_const(path=p, basename=fname, handler_rec=handler_rec)
+            # FIXME: we might have to deal with UTF-8 decoding issues
+            # reader = csv.reader(codecs.iterdecode(tb.extractfile(dfilepath), 'utf-8'))
+            datafile['reader'] = reader = csv.reader(open(os.path.join(ptb.extracted_root, p)))
+            # FIXME: how should we handle .csv files that are empty?
+            datafile['header'] = next(reader)
+            datafiles.append(datafile)
+        return datafiles
+
+    @staticmethod
+    def get_json_files(handler, iteration, sample, host, tool, toolgroup, ptb):
+        """
+        Fetch the list of json files for this tool, and
+        return a list of dicts containing their metadata.
+        """
+        # import pdb; pdb.set_trace()
+        path = os.path.join(iteration, sample, "tools-%s" % (toolgroup,), host, tool, "json")
+        paths = [x for x in ptb.tb.getnames() if x.find(path) >= 0 and ptb.tb.getmember(x).isfile()]
+        datafiles = []
+        for p in paths:
+            fname = os.path.basename(p)
+            datafile = _dict_const(path=p, basename=fname, handler_rec=None)
+            datafiles.append(datafile)
+        return datafiles
 
-# check that all the directories exist
-test -d $ARCHIVE || doexit "Bad ARCHIVE=$ARCHIVE"
-
-log_init $PROG
-
-# make sure only one copy is running.
-# Use 'flock -n $LOCKFILE /home/pbench/bin/pbench-index' in the
-# crontab to ensure that only one copy is running. The script itself
-# does not use any locking.
-
-# the link source and destination for this script
-linksrc=TO-INDEX
-linkdest=INDEXED
-linkerrdest=WONT-INDEX
-
-# get the list of files we'll be operating on
-#list=$(ls $ARCHIVE/*/$linksrc/*.tar.xz 2>/dev/null)
-# Better: avoids long command line problems and sorts by size
-TMPDIR=$TMP/pbench-index.$$
-# index-pbench need to get at it
-export TMPDIR
-
-echo "$TS: starting at $(timestamp)"
-
-typeset -i nidx=0
-indexed=$TMPDIR/$PROG.$TS.indexed
-typeset -i nerrs=0
-erred=$TMPDIR/$PROG.$TS.erred
-typeset -
```

Comment:
Is this should be "logger" instead of "_logger" because you have used `logger.*` everywhere else in the code Or am I missing something here?

Reply:
I think we want it to be `_logger` so that we use the convention of variables marked with a leading underscore denoting module local, not to be exported.